src/raftstore/store/worker/apply.rs

```diff
@@ -2001,12 +2163,19 @@ impl Runner {
         notifier: Sender<TaskRes>,
         sync_log: bool,
         use_delete_range: bool,
+        ch: Scheduler<Task>,
     ) -> Runner {
         let mut delegates =
             HashMap::with_capacity_and_hasher(store.get_peers().len(), Default::default());
         for (&region_id, p) in store.get_peers() {
             delegates.insert(region_id, Some(ApplyDelegate::from_peer(p)));
         }
+
+        // Start sub_apply_worker to handle time cost request asynchronously.
+        let mut sub_apply_worker = Worker::new("sub apply worker");
+        let sub_apply_runner = SubApplyRunner::new(store.kv_engine(), use_delete_range, ch);
+        sub_apply_worker.start(sub_apply_runner).unwrap();
```

Comment:
Is it proper to start the sub runner here, in the `new` method of Runner? The outer runner is not started at this place. I think you'd better start the sub runner when the outer runner is started.